### PR TITLE
Docs for wildcard policy subdomain routes.

### PR DESCRIPTION
**Pull request by ramr**
docs for enabling wildcard routes (routes with a wildcard policy of subdomain).
Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1384760

 @ahardin-rh / @adellape / @bfallonf  / @knobunc  PTAL  Thx

### CODE REVIEW

File: architecture/core_concepts/routes.adoc [hunk 1]
```diff
@@ -802,3 +802,34 @@ If unit not provided, ms is the default.
...
 WebSocket connections to timeout frequently on that route.
 ====
+
+
+[[wildcard-subdomain-route-policy]]
+== Creating routes specifying a wildcard subdomain policy
+
+A route can specify a wildcard policy as part of its configuration using the
+`wildcardPolicy` field. Any routers which are run with a policy that allows
+wildcard routes, will expose the route appropriately based on the wildcard
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: the comma is not necessary

File: architecture/core_concepts/routes.adoc [hunk 2]
```diff
@@ -802,3 +802,34 @@ If unit not provided, ms is the default.
 Setting a server-side timeout value for passthrough routes too low can cause
 WebSocket connections to timeout frequently on that route.
 ====
+
+
+[[wildcard-subdomain-route-policy]]
+== Creating routes specifying a wildcard subdomain policy
+
+A route can specify a wildcard policy as part of its configuration using the
```
Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: Can we say what they do as a first sentence?

"A wildcard policy allows a user to define a route that can cover all hosts within a domain (when the router is configured to allow it)."

Or something

File: install_config/router/default_haproxy_router.adoc [hunk 3]
```diff
@@ -824,6 +824,88 @@ DNS lookup:
...
+DNS lookup:
+
+----
+# routerip="4.1.1.1"  #  replace with IP address of one of your router instances.
+# curl -k --resolve www.example.test:443:$routerip https://www.example.test/
+# curl -k --resolve abc.example.test:443:$routerip https://abc.example.test/
+# curl -k --resolve anyname.example.test:443:$routerip https://anyname.example.test/
+----
+
+
```
Comment (knobunc) [inline install_config/router/default_haproxy_router.adoc]: Nice example.  Thanks!

**Review by knobunc — DISMISSED** [review]

File: architecture/core_concepts/routes.adoc [hunk 4]
```diff
@@ -802,3 +802,34 @@ If unit not provided, ms is the default.
 Setting a server-side timeout value for passthrough routes too low can cause
 WebSocket connections to timeout frequently on that route.
 ====
+
+
+[[wildcard-subdomain-route-policy]]
+== Creating routes specifying a wildcard subdomain policy
```
Comment (ahardin-rh) [inline architecture/core_concepts/routes.adoc]: put headings in title case:
Creating Routes Specifying a Wildcard Subdomain Policy

File: architecture/core_concepts/routes.adoc [hunk 5]
```diff
@@ -802,3 +802,34 @@ If unit not provided, ms is the default.
...
+
+A route can specify a wildcard policy as part of its configuration using the
+`wildcardPolicy` field. Any routers which are run with a policy that allows
+wildcard routes, will expose the route appropriately based on the wildcard
+policy.
+To configure HAProxy routers to allow wildcard routes, see
+xref:../../install_config/router/default_haproxy_router.adoc#using-wildcard-routes[using wildcard routes with HAProxy]
+
+
+.A Route Specifying a subdomain WildcardPolicy
```
Comment (ahardin-rh) [inline architecture/core_concepts/routes.adoc]: s/subdomain/Subdomain

**Comment by ahardin-rh** [comment]
@ramr Thank you! I have a few minor style edits to apply, which I will take care of in a follow-up PR. I will also address the feedback from @knobunc in there as well. With that said, I will merge this now and tag you and @knobunc in my follow-up.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../architecture/core_concepts/routes.adoc#architecture-core-concepts-routes[Core Concepts -> Routes]
|Added a new xref:../architecture/core_concepts/routes.adoc#wildcard-subdomain-route-policy[Creating Routes Specifying a Wildcard Subdomain Policy] section.
%
|xref:../install_config/router/default_haproxy_router.adoc#install-config-router-default-haproxy[Setting up a Router -> Using the Default HAProxy Router]
|Added a new xref:../install_config/router/default_haproxy_router.adoc#using-wildcard-routes[Using Wildcard Routes (for a Subdomain)] section.
%

**Comment by ramr** [comment]
@ahardin-rh  Thx. 
Am back online now and I did want to add a note/blurb about ownership/security implications - which I'll do sometime later this afternnon in a new PR (need to get my email sorted before that). So I can address the review comments in there as well.

**Comment by ramr** [comment]
nevermind the previous comment - saw the follow up PR: #3234 - I'll just create PR for the note/blurb. thx

**Pull request merged**